### PR TITLE
Update rd-cpp from JetBrains/rd

### PR DESCRIPTION
**Pull request by LeoMash**

### CODE REVIEW

**Comment by DecoyRS** [comment]
134 files. SHEEESH

**Comment by LeoMash** [comment]
More than half of it is updated spdlog.
Spdlog update was needed to have updated bundled fmt, as the old one was incompatible with msvc c++20.

File: src/cpp/RiderLink/Source/RD/src/rd_framework_cpp/src/main/task/WiredRdTaskImpl.h [hunk 1]
```diff
@@ -44,8 +44,9 @@ class WiredRdTaskImpl : public RdReactiveBase
 	void on_wire_received(Buffer buffer) const override
 	{
 		auto read_result = RdTaskResult<T, S>::read(cutpoint->get_serialization_context(), buffer);
-		spdlog::get("logReceived")->trace("call {} {} received response {} : {}", to_string(cutpoint->location), to_string(rdid), to_string(rdid),
-			to_string(read_result));
+		spdlog::get("logReceived")
+			->trace("call {} {} received response {} : {}", to_string(cutpoint->get_location()), to_string(rdid), to_string(rdid),
```
Comment (DecoyRS) [inline src/cpp/RiderLink/Source/RD/src/rd_framework_cpp/src/main/task/WiredRdTaskImpl.h:48]: Split the lines at `->` operator ? That's smth new

File: src/cpp/RiderLink/Source/RD/src/rd_framework_cpp/src/main/util/hashing.h [hunk 2]
```diff
@@ -30,7 +30,7 @@ constexpr hash_t getPlatformIndependentHash(char const (&that)[N], constexpr_has
 
 constexpr hash_t getPlatformIndependentHash(string_view that, constexpr_hash_t initial = DEFAULT_HASH)
 {
-	return static_cast<hash_t>(hashImpl(initial, &that[0], &that[that.length()]));
+	return static_cast<hash_t>(hashImpl(initial, &that[0], &that[that.length() - 1] + 1));
```
Comment (DecoyRS) [inline src/cpp/RiderLink/Source/RD/src/rd_framework_cpp/src/main/util/hashing.h:33]: Can you explain what's happening here so I don't need to open the source code? :) We're taking an address "next after the last one"?

File: src/cpp/RiderLink/Source/RD/src/rd_framework_cpp/src/main/base/RdBindableBase.h [hunk 3]
```diff
@@ -54,27 +63,24 @@ class RD_FRAMEWORK_API RdBindableBase : public virtual IRdBindable /*, IPrintabl
 	// mutable std::map<std::string, std::any> non_bindable_extensions;//TO-DO concurrency
 
 	template <typename T, typename... Args>
-	typename std::enable_if_t<util::is_base_of_v<IRdBindable, T>, T> const& getOrCreateExtension(
-		std::string name, Args&&... args) const
+	auto getOrCreateExtension(std::string name, Args&&... args) const ->
+		typename std::enable_if_t<util::is_base_of_v<IRdBindable, T>, T> const&
```
[the remark this answers is not in the document]
Comment (LeoMash) [reply src/cpp/RiderLink/Source/RD/src/rd_framework_cpp/src/main/base/RdBindableBase.h:67]: Stylistic change. I'm lost when such long type with SFINAE is before declaration. Also during fixes there was another change  that was reverted (I tried to remove const, but failed)

File: src/cpp/RiderLink/Source/RD/src/rd_framework_cpp/src/main/util/hashing.h (hunk 2 above)
Comment (LeoMash) [reply src/cpp/RiderLink/Source/RD/src/rd_framework_cpp/src/main/util/hashing.h:33]: &that[that.length()] - is the end pointer, i.e. invalid pointer after string.
In debug string_view::operator[] has debug checks if index is out of range and when evaluating in constexpr context such access to end pointer cause compilation error.
So I changed retrieval of end pointer as pointer to last element + 1

**Review by DecoyRS — APPROVED** [review]
lgtm

**Pull request merged**